src/general/VersionUtils.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using Godot;
+/// <summary>
+///   Helpers for dealing with Thrive's version number
+/// </summary>
+public static class VersionUtils
+{
+    /// <summary>
+    /// Compare the given version numbers.
+    /// </summary>
+    /// <returns>
+    /// 0 if the versions are the same,
+    /// a negative integer if a is a smaller (older) version than b and
+    /// a positive integer if a is a bigger (newer) version than b
+    /// </returns>
+    /// <param name="a">The first version to compare.</param>
+    /// <param name="b">The second version to compare.</param>
+    public static int Compare(string a, string b)
+    {
+        if (a == b)
+        {
+            return 0;
+        }
+
+        char[] separators = { '.', '-' };
+        var aSplit = a.Split(separators);
+        var bSplit = b.Split(separators);
+
+        for (int i = 0; i < aSplit.Length; i++)
```

Comment:
What about the case where bSplit.length < aSplit.length?

Reply:
Maybe a library like https://github.com/maxhauser/semver would be better then?

Reply:
Yes, that looks pretty nice and is licensed under the MIT license. If you add that remember to modify the Thrive license to mention the new used library.

Reply:
I hope that works for thrive as we have 4 components in version numbers instead of just 3.
But that is from the standard C# assembly version so it should be doable to find a library to compare those.

Reply:
...except that Thrive version is not semver (one number too many).

Reply:
Looks like the standard C# library might have something: https://docs.microsoft.com/en-us/dotnet/api/system.version.parse?view=netcore-3.1
didn't test in mono if that is available.

Reply:
Though that doesn't seem to handle suffixes, so maybe splitting on '-' and using Version to parse the first part and string comparison (or hard coded values like alpha, beta, pre-alpha) to determine the more accurate order if the Version is the same.

Reply:
Can I assume that a version only has exactly one or no '-' and having one means it's earlier than one without it? Failing these assumption could give it's own warning message.

Reply:
That's not a safe assumption.
0.5.2.0-pre-alpha is a valid version that needs to be handled.
Shame that we don't have unit testing setup to make it easy to verify that the version comparisons work.

Reply:
Pre-alpha will also not be comparable alphabetically. How about having a predefined set of suffixes (including empty) and failing if none match?

Reply:
I'd prefer to also have ordinal comparison fallback.
But a list of suffixes is a good idea, here's what I have of the top of my head:
```
pre-alpha
alpha
beta
rc1
rc2
rc3
rcn
```